### PR TITLE
Fix/toast exceeded limit channels created

### PR DESCRIPTION
**Pull request by giachuvan**

### CODE REVIEW

File: crates/mezon-store/src/threads.rs [hunk 1]
```diff
@@ -576,6 +587,22 @@ impl ThreadsStore {
...
 
+    fn invalidate_create_request(&mut self) {
+        self.create_request_generation = self.create_request_generation.wrapping_add(1);
+        self._create_task = None;
+        self.submitting = false;
+    }
+
+    fn finish_create_request(&mut self, generation: u64) -> bool {
+        if self.create_request_generation != generation {
+            self._create_task = None;
```
Comment (huylexuan44) [inline crates/mezon-store/src/threads.rs]: Here _create_task can only hold a newer task, so clearing it cancels an unrelated in-flight create. Remove this line.

File: crates/mezon-store/src/threads.rs [hunk 2]
```diff
@@ -1007,8 +1045,13 @@ impl ThreadsStore {
             }
 
             let _ = this.update(cx, |this, cx| {
+                if this.create_request_generation != generation {
+                    this._create_task = None;
```
Comment (huylexuan44) [inline crates/mezon-store/src/threads.rs]: Same problem as line 598: this cancels a newer create task on a stale generation. Remove.

File: crates/mezon-store/src/threads.rs [hunk 3]
```diff
@@ -576,6 +587,22 @@ impl ThreadsStore {
         self.clan_id.as_ref().filter(|id| !id.is_empty()).cloned()
     }
 
+    fn invalidate_create_request(&mut self) {
```
Comment (huylexuan44) [inline crates/mezon-store/src/threads.rs:586]: create_request_generation is redundant with _create_task: every bump already drops the old task. Consider removing.

File: crates/mezon-store/src/threads.rs [hunk 4]
```diff
@@ -961,9 +995,11 @@ impl ThreadsStore {
                 Err(e) => {
                     tracing::error!("create_channel (thread) failed: {e}");
                     let _ = this.update(cx, |this, cx| {
-                        this.submitting = false;
+                        if !this.finish_create_request(generation) {
```
Comment (huylexuan44) [inline crates/mezon-store/src/threads.rs]: let _ on this.update hides a failure that now leaves _create_task set and blocks the next submit. Log the error.

File: crates/mezon-client/src/transport.rs [hunk 5]
```diff
@@ -6103,7 +6128,7 @@ impl MezonTransport {
             .await?;
 
         if code != 0 {
-            return Err(anyhow::anyhow!("API error: code={}", code));
+            return Err(api_status_error(code));
```
Comment (huylexuan44) [inline crates/mezon-client/src/transport.rs:6143]: Code 11 is generic OUT_OF_RANGE. Any other out-of-range failure on CreateChannelDesc now shows the channel-limit toast.

File: assets/i18n/en.json [hunk 6]
```diff
@@ -2111,6 +2111,7 @@
   "common.uploadLimit.emoji": "Max emoji limit exceeded",
   "common.uploadLimit.sticker": "Max sticker limit exceeded",
   "common.uploadLimit.voiceSticker": "Max voice sticker limit exceeded",
+  "common.uploadLimit.channel": "Maximum channel limit reached for this clan",
```
Comment (huylexuan44) [inline assets/i18n/en.json]: uploadLimit is a React-mirrored namespace for uploads. Use a desktop-only key, e.g. channel.createLimitExceeded.

File: crates/mezon-ui/src/components/primitives/toast.rs [hunk 7]
```diff
@@ -66,8 +66,8 @@ impl RenderOnce for Toast {
             .py(px(10.))
             .rounded_md()
             .border_1()
-            .border_color(theme.border)
-            .bg(theme.bg_floating)
+            .border_color(theme.tokens.border_primary)
+            .bg(theme.bg_primary)
```
Comment (huylexuan44) [inline crates/mezon-ui/src/components/primitives/toast.rs]: bg_floating to bg_primary is a real color change and unrelated to this PR. Every other shell overlay uses bg_floating.

File: crates/mezon-ui/src/components/primitives/toast.rs [hunk 8]
```diff
@@ -66,8 +66,8 @@ impl RenderOnce for Toast {
             .py(px(10.))
             .rounded_md()
             .border_1()
-            .border_color(theme.border)
-            .bg(theme.bg_floating)
+            .border_color(theme.tokens.border_primary)
```
Comment (huylexuan44) [inline crates/mezon-ui/src/components/primitives/toast.rs]: from_tokens already sets theme.border = tokens.border_primary, so this line is a no-op. Drop it or explain.

File: crates/mezon-ui/src/chat/layout.rs [hunk 9]
```diff
@@ -293,10 +294,9 @@ impl ChatLayout {
             }
         })
         .detach();
-        cx.subscribe(&threads_store, |this, _, event, cx| {
+        let threads_event_sub = cx.subscribe(&threads_store, |this, _, event, cx| {
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/layout.rs]: ChatLayout is built once in RootView::new, so .detach() was already correct. Converting 1 of 19 subs is inconsistent.

File: crates/mezon-ui/src/chat/layout.rs [hunk 10]
```diff
@@ -1959,7 +1960,8 @@ impl ChatLayout {
                 self.navigate_to_thread(channel_id, clan_id, "", "", cx);
                 ThreadsStore::global(cx).update(cx, |store, cx| store.refresh(cx));
             }
-            ThreadsEvent::CreateFailed { message } | ThreadsEvent::LeaveFailed { message } => {
+            ThreadsEvent::CreateFailed { .. } => {}
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/layout.rs]: CreateFailed is now typed but LeaveFailed still toasts raw e.to_string(). Same enum, two conventions.

File: crates/mezon-store/src/threads.rs [hunk 11]
```diff
@@ -1020,9 +1063,18 @@ impl ThreadsStore {
...
                 cx.notify();
             });
-        })
-        .detach();
+        });
+        self._create_task = Some(task);
+    }
+}
+
+fn thread_create_fail_reason(err: &anyhow::Error) -> ThreadCreateFailReason {
```
Comment (huylexuan44) [inline crates/mezon-store/src/threads.rs:1056]: Same predicate as map_create_channel_api_error in channel.rs:231. Extract one helper or ApiStatusError::is_out_of_range.

File: crates/mezon-store/src/threads.rs [hunk 12]
```diff
@@ -6,7 +6,8 @@ use mezon_client::AppApi;
 use mezon_client::ConnectionStatus;
 use mezon_client::MezonTransport;
 use mezon_client::RealtimeEvent;
-use mezon_client::transport::{ApiThreadDesc, THREAD_LIST_LIMIT};
+use mezon_client::api_status_from_error;
```
Comment (huylexuan44) [inline crates/mezon-store/src/threads.rs]: api_status_from_error comes from mezon_client:: but ApiStatusError from mezon_client::transport::. Pick one path.

File: crates/mezon-ui/src/clan/create_channel_modal.rs [hunk 13]
```diff
@@ -171,6 +171,15 @@ impl CreateChannelModal {
                     cx.notify();
                 });
             }
+            Err(CreateChannelError::ChannelLimitExceeded) => {
```
Comment (huylexuan44) [inline crates/mezon-ui/src/clan/create_channel_modal.rs:175]: InvalidName and DuplicateName render inline Validation; this shows a toast. Add Validation::ChannelLimit instead.

File: crates/mezon-ui/src/app/shell/mod.rs [hunk 14]
```diff
@@ -164,6 +164,36 @@ impl Shell {
         self.toast(ToastKind::Error, message, cx);
     }
 
+    pub fn error_once(
```
Comment (huylexuan44) [inline crates/mezon-ui/src/app/shell/mod.rs:167]: error_once duplicates toast(). Call self.toast() then set key on self.toasts.last_mut().

File: crates/mezon-client/src/lib.rs [hunk 15]
```diff
@@ -71,6 +71,7 @@ pub use transport::{
     CANVAS_LIST_LIMIT, CANVAS_STATUS_CREATED, CANVAS_STATUS_UPDATE, HttpFallbackSession,
     parse_search_attachment_field, parse_search_mentions_field,
 };
+pub use transport::{ApiStatusError, api_status_from_error};
```
Comment (huylexuan44) [inline crates/mezon-client/src/lib.rs]: Fold into the pub use transport::{...} block directly above instead of adding a second one.

**Comment by huylexuan44** [comment]
Review summary:

- Holding the create task in _create_task is the real root-cause fix: cancel_create previously cleared submitting while leaving a detached task running, so a resubmit produced two CreateFailed events. Good catch.
- The generation counter on top of it is redundant, and its mismatch branches clear _create_task when that field can only hold a newer task. See inline comments on threads.rs:590/598/1049.
- No tests. channel.rs has 111 and threads.rs has 14, and map_create_channel_api_error, thread_create_fail_reason and api_status_from_error are pure functions. The submit/cancel/resubmit transition this PR fixes also deserves one.
- The Toast restyle and the ChatLayout subscription change are unrelated to the stated fix. Split them out or note them in the PR description.
